src/StoreApi/Routes/Checkout.php

```diff
@@ -461,7 +461,7 @@ private function update_customer_from_request( \WP_REST_Request $request ) {
 	 */
 	private function update_order_from_request( \WP_REST_Request $request ) {
 		$this->order->set_customer_note( $request['customer_note'] ?? '' );
-		$this->order->set_payment_method( $this->order->needs_payment() ? $this->get_request_payment_method( $request ) : '' );
+		$this->order->set_payment_method( $this->get_request_payment_method( $request ) );
```

Comment:
Initially I just thought I should keep the diff small as I'm not familiar with the codebase, but apparently the function `get_request_payment_method` is only being used here.

We could perhaps replace it for `get_request_payment_method_id` and remove the other one.